CHANGELOG.md

```diff
@@ -1,5 +1,20 @@
 # Change Log
 
+# [1.28.0 (September 28th, 2023)](https://github.com/eclipse/eclipse.jdt.ls/milestone/119?closed=1)
+ * enhancement - Update current method snippet and add `static_method` snippet for interface. See [#2862](https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2862).
+ * enhancement - Improve the constructor snippet for additional classes in a file. See [#2841](https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2841).
+ * enhancement - Return outline/documentSymbols even for `.java` files that are not in a Java project. See [#2809](https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/2809).
+ * enhancement - Store the completion kinds requested by completion operation. See [#2857](https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2857).
+ * enhancement - Report whether the project has Kotlin Gradle files. See [#2859](https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2859).
+ * bug fix - Fix `EmptyStackException` in `textDocument/foldingRange`. See [#2865](https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/2865).
+ * bug fix - Fix renaming of attribute with `@Singular` annotation. See [#2828](https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2828).
+ * bug fix - Fix `URI` with query parameter. See [#2861](https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2861).
+ * bug fix - Fix an error thrown during "Initialize workspace". See [#2842](https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/2842).
+ * bug fix - Fix potential for NPE in `JDTUtils.getClassFile()`. See [#2873](https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2873).
```

Comment:
typos: `for`?

Reply:
I guess I could have just said "Fix potential NPE ..." . "potential for NPE" / "potnetial of NPE" / "potential NPE" don't really change the meaning here so I think it's best to just go for least amount of words.

Reply:
Great.  I have learned something new. thanks for the explanation.